validate_output.py

```diff
@@ -3,32 +3,49 @@
 def GetAuthorsList(mList):
     finalDic = defaultdict(int)
     for x in mList:
-        finalDic[x[0]] = None  
+        finalDic[x[0]] = 0  
     return finalDic
 
-def validate(inList, outList, maxPerAuthor):
+def Validate(inList, outList, maxPerAuthor, verbose):
     authors = GetAuthorsList(inList)
     
     meetsReq = True
-    numberOfAuthorsWithNoPapers = 0
-    numberOfAuthorsWithToManyPapers = 0
+    authorsWithNoPapers = []
+    authorsWithTooManyPapers = []
+    authorsWithMultiplePapers = []
     
     for x in outList:
-        authors[x[1]] += 1
+        authors[x[0]] += 1
 
     for y in authors:
         if authors[y] == 0:
             meetsReq = False
-            numberOfAuthorsWithNoPapers += 1
+            authorsWithNoPapers.append(y)
         if authors[y] > maxPerAuthor:
             meetsReq = False
-            numberOfAuthorsWithToManyPapers += 1
-    
-    print(f"Valid results?: {meetsReq}")
-    print(f"Total errors: {numberOfAuthorsWithNoPapers + numberOfAuthorsWithToManyPapers}")
-    print(f"Total number of authors with no papers: {numberOfAuthorsWithNoPapers}")
-    print(f"Total number of authors with to many papers: {numberOfAuthorsWithNoPapers}")
-
-
+            authorsWithTooManyPapers.append(y)
 
+    for a in authors:
+        if authors[a] > 1:
+            authorsWithMultiplePapers.append(a)
+    
+    numberOfAuthorsWithNoPapers = len(authorsWithNoPapers)
+    numberOfAuthorsWithTooManyPapers = len(authorsWithTooManyPapers)
+    numberOfAuthorsWithMultiplePapers = len(authorsWithMultiplePapers)
 
+    if verbose == True:
+        print()
+        print(f"Valid results?: {meetsReq}")
+        print(f"Total errors: {numberOfAuthorsWithNoPapers + numberOfAuthorsWithTooManyPapers}")
+        print(f"Authors with no papers: {str(authorsWithNoPapers)}")
+        print(f"Total number of authors with no papers: {numberOfAuthorsWithNoPapers}")
+        print(f"Authors with too many papers: {str(authorsWithTooManyPapers)}")
+        print(f"Total number of authors with too many papers: {numberOfAuthorsWithTooManyPapers}")
+        print(f"Authors with multiple papers: {str(authorsWithMultiplePapers)}")
```

Comment:
In cases like this i dont think you have to cast variable to string, so its cleaner if you dont.